rust/src/combinators/select_private_candidate/mod.rs

```diff
@@ -0,0 +1,143 @@
+use crate::{
+    core::{Domain, Function, Measurement, Metric, MetricSpace, PrivacyMap},
+    error::Fallible,
+    measures::MaxDivergence,
+    traits::{
+        samplers::sample_geometric_exp_fast, CastInternalRational, InfAdd, InfCast, InfDiv, InfExp,
+        InfLn, InfMul, InfSub, NextFloat,
+    },
+};
+use dashu::integer::UBig;
+use opendp_derive::bootstrap;
+use std::{f64::consts::E, fmt::Debug, ops::Neg};
+
+#[cfg(feature = "ffi")]
+mod ffi;
+
+#[cfg(test)]
+mod test;
+
+#[bootstrap(
+    features("contrib"),
+    arguments(
+        measurement(rust_type = "AnyMeasurement"),
+        max_iterations(default = b"null")
+    ),
+    generics(DI(suppress), MI(suppress), TO(suppress)),
+    dependencies("$get_dependencies(measurement)")
+)]
+/// Select a private candidate whose score is above a threshold.
+///
+/// Given `measurement` that satisfies ε-DP, returns new measurement M' that satisfies 2ε-DP.
+/// M' releases the first invocation of `measurement` whose score is above `threshold`.
+///
+/// Each time a score is below `threshold`
+/// the algorithm may terminate with probability `stop_probability` and return nothing.
+///
+/// `measurement` should make releases in the form of (score, candidate).
+/// If you are writing a custom scorer measurement in Python,
+/// specify the output type as `TO=(float, "ExtrinsicObject")`.
+/// This ensures that the float value is accessible to the algorithm.
+/// The candidate, left as arbitrary Python data, is held behind the ExtrinsicObject.
+///
+/// Algorithm 1 in `Private selection from private candidates <https://arxiv.org/pdf/1811.07971.pdf#page=7>`_ (Liu and Talwar, STOC 2019).
+///
+/// # Arguments
+/// * `measurement` - A measurement that releases a 2-tuple of (score, candidate)
+/// * `stop_probability` - The probability of stopping early at any iteration.
+/// * `threshold` - The threshold score. Return immediately if the score is above this threshold.
+/// * `max_iterations` - Optional. Run the algorithm at most this many iterations.
+///
+/// # Returns
+/// A measurement that returns a release from `measurement` whose score is greater than `threshold`, or none.
```

Comment:
Python usage example might be included? I guess I'm still leaning towards a little more documentation now, while it's on your mind.